ledger/state.go

```diff
@@ -1033,4 +1224,34 @@ func (ls *LedgerState) forgeBlock() {
 		"prev_hash", hex.EncodeToString(currentTip.Point.Hash),
 		"block_cbor", hex.EncodeToString(blockCbor),
 	)
+
+	// Create a proper ledger.Block from the Conway block
+	ledgerBlock := ledger.Block(conwayBlock)
+
+	// Add the block to the primary chain
+	txn := ls.db.Transaction(true)
+	err = txn.Do(func(txn *database.Txn) error {
+		return ls.chain.AddBlock(ledgerBlock, txn)
+	})
```

Comment:
There's no need to provide a DB transaction here. The underlying function will provide its own if you don't provide one, and it won't lock the sqlite DB like this will.